bench/topic_mysql_perf.rb

```diff
@@ -0,0 +1,310 @@
+    s << t.id.to_s
+    s << t.title
+  end
+  s
+end
+
+def sequel_select_title_id
+  s = +""
+  TopicSequel.limit(1000).order(:id).select(:id, :title).each do |t|
+    s << t.id.to_s
+    s << t.title
+  end
+  s
+end
+
+def sequel_pluck_title_id
+  s = +""
+  TopicSequel.limit(1000).order(:id).select_map([:id, :title]).each do |t|
+    s << t[0].to_s
+    s << t[1]
+  end
+  s
+end
+
+# usage is not really recommended but just to compare to pluck lets have it
+def mini_sql_title_id_query_single
+  s = +""
+  i = 0
+  r = $mini_sql.query_single(-"select id, title from topics order by id limit 1000")
+  while i < r.length
+    s << r[i].to_s
+    s << r[i+1]
+    i += 2
+  end
+  s
+end
+
+results = [
+  ar_title_id,
+  ar_title_id_pluck,
+  mysql_title_id,
+  mini_sql_title_id,
+  sequel_pluck_title_id,
+  sequel_select_title_id,
+  mini_sql_title_id_query_single
+]
+
+exit(-1) unless results.uniq.length == 1
+
+
+Benchmark.ips do |r|
+  r.report("ar select title id") do |n|
+    while n > 0
+      ar_title_id
+      n -= 1
+    end
+  end
+  r.report("ar select title id pluck") do |n|
+    while n > 0
+      ar_title_id_pluck
+      n -= 1
+    end
+  end
+  r.report("sequel title id select") do |n|
+    while n > 0
+      sequel_select_title_id
+      n -= 1
+    end
+  end
+  r.report("mysql select title id") do |n|
+    while n > 0
+      mysql_title_id
+      n -= 1
+    end
+  end
+  r.report("mini_sql select title id") do |n|
+    while n > 0
+      mini_sql_title_id
+      n -= 1
+    end
+  end
+  r.report("sequel title id pluck") do |n|
+    while n > 0
+      sequel_pluck_title_id
+      n -= 1
+    end
+  end
+  r.report("mini_sql query_single title id") do |n|
+    while n > 0
+      mini_sql_title_id_query_single
+      n -= 1
+    end
+  end
+  r.compare!
+end
+
+
+
+def wide_topic_ar
+  Topic.first
+end
+
+def wide_topic_mysql
+  r = $conn.query("select * from topics limit 1", as: :hash)
+  row = r.first
+  row
+end
+
+def wide_topic_sequel
+  TopicSequel.first
+end
+
+def wide_topic_mini_sql
+  $conn.query("select * from topics limit 1").first
+end
+
+Benchmark.ips do |r|
+  r.report("wide topic ar") do |n|
+    while n > 0
+      wide_topic_ar
+      n -= 1
+    end
+  end
+  r.report("wide topic sequel") do |n|
+    while n > 0
+      wide_topic_sequel
+      n -= 1
+    end
+  end
+  r.report("wide topic mysql") do |n|
+    while n > 0
+      wide_topic_mysql
+      n -= 1
+    end
+  end
+  r.report("wide topic mini sql") do |n|
+    while n > 0
+      wide_topic_mini_sql
+      n -= 1
+    end
+  end
+  r.compare!
+end
+
+# Comparison:
+# mysql select title id:      486.5 i/s
+# mini_sql query_single title id:      450.3 i/s - same-ish: difference falls within error
+# sequel title id pluck:      367.9 i/s - 1.32x  slower
+# sequel title id select:      352.5 i/s - 1.38x  slower
+# mini_sql select title id:      346.2 i/s - 1.41x  slower
```

Comment:
I am a bit concerned about this number is there anything we can do to improve this?

Reply:
improved 😃